tool/gravity/cli/config.go

```diff
@@ -865,12 +868,42 @@ type agentConfig struct {
 	cloudProvider string
 }
 
+func retryUpdateJoinConfigFromCloudMetadata(ctx context.Context, config *autojoinConfig) error {
+	// Security: Insecure is OK here because we're only testing reachability to the cluster
+	client := httplib.GetClient(true,
+		httplib.WithTimeout(5*time.Second),
+		httplib.WithInsecure())
+
+	b := backoff.NewConstantBackOff(15 * time.Second)
+	f := func() error {
+		err := updateJoinConfigFromCloudMetadata(ctx, config)
+		if err != nil && !trace.IsRetryError(err) {
+			return &backoff.PermanentError{Err: err}
+		}
+
+		// Test that the serviceURL is reachable
+		// When re-installing a cluster into AWS, the serviceURL can point to an old cluster
+		// until the new cluster overwrites the SSM values. So test the ServiceURL is reachable before using it.
+		req, _ := http.NewRequest("GET", config.serviceURL, nil)
+		req = req.WithContext(ctx)
```

Comment:
There's [NewRequestWithContext](https://golang.org/pkg/net/http/#NewRequestWithContext) - or did you want to avoid it for some reason?
Also I'm not sure about ignoring an error here - is it guaranteed to succeed?

Reply:
NewRequestWithContext was added in go 1.13, I think we're still using go 1.12.

Reply:
NewRequestWithContext was added in go1.13, I think we're still on golang 1.12 on this repo.